File: test/java/com/marsrovers/control/CommandExecutorTest.java
package com.marsrovers.control;

import com.marsrovers.location.Orientation;
import com.marsrovers.location.Position;
import org.junit.Test;
import com.marsrovers.units.Rover;

import static org.junit.Assert.assertEquals;


public class CommandExecutorTest {

    private final CommandExecutor executor = new CommandExecutor();
    private final Rover rover = new Rover("a", Orientation.NORTH, new Position(1,1));

    @Test
    public void executeShouldSkipNonExecutableCommand() throws Exception {
        Command command = new Command("I am not executable");

        Rover movedRover = executor.execute(rover, command);

        assertEquals(rover, movedRover);
    }

    @Test
    public void executeShouldTurnRoverLeft() throws Exception {


        Command command = new Command("L");

        Rover movedRover = executor.execute(rover, command);

        assertEquals(Orientation.WEST, movedRover.orientation);

    }

    @Test
    public void executeShouldMoveRover() throws Exception {

        Command command = new Command("M");

        Rover movedRover = executor.execute(rover, command);

        assertEquals(2, movedRover.position.getY());

    }

    @Test
    public void executeShouldTurnRoverRight() throws Exception {

        Command command = new Command("R");

        Rover movedRover = executor.execute(rover, command);

        assertEquals(Orientation.EAST, movedRover.orientation);

    }

    @Test
    public void executeShouldExecuteMultipleMovesCommand() throws Exception {

        Command command = new Command("LLM");

        Rover movedRover = executor.execute(rover, command);

        assertEquals(0, movedRover.position.getY());

    }
}

File: test/java/com/marsrovers/control/CommandExecutorTest.kt
package com.marsrovers.control

import com.marsrovers.location.Orientation
import com.marsrovers.location.Position
import org.junit.Test
import com.marsrovers.units.Rover

import org.junit.Assert.assertEquals


class CommandExecutorTest {

    private val executor = CommandExecutor()
    private val rover = Rover("a", Orientation.NORTH, Position(1, 1))

    @Test
    @Throws(Exception::class)
    fun executeShouldSkipNonExecutableCommand() {
        val command = Command("I am not executable")

        val movedRover = executor.execute(rover, command)

        assertEquals(rover, movedRover)
    }

    @Test
    @Throws(Exception::class)
    fun executeShouldTurnRoverLeft() {


        val command = Command("L")

        val movedRover = executor.execute(rover, command)

        assertEquals(Orientation.WEST, movedRover.orientation)

    }

    @Test
    @Throws(Exception::class)
    fun executeShouldMoveRover() {

        val command = Command("M")

        val movedRover = executor.execute(rover, command)

        assertEquals(2, movedRover.position.y.toLong())

    }

    @Test
    @Throws(Exception::class)
    fun executeShouldTurnRoverRight() {

        val command = Command("R")

        val movedRover = executor.execute(rover, command)

        assertEquals(Orientation.EAST, movedRover.orientation)

    }

    @Test
    @Throws(Exception::class)
    fun executeShouldExecuteMultipleMovesCommand() {

        val command = Command("LLM")

        val movedRover = executor.execute(rover, command)

        assertEquals(0, movedRover.position.y.toLong())

    }
}